### ISSUE TITLE
Swapping between maps jiggles the selected layer and groups

### ISSUE THREAD

**Issue by Macksaur** [mapeditor/tiled#3282]
### Problem

When I am moving around multiple maps and copy/pasting pieces as source material and I swap back to the target map I find all the groups closed and the layer I was on forgotten and instead replaced with a "best guess" at a layer with the same name.

```
Map A:
- Group1
    - Objects
    - ActualTarget <-- this layer is selected
- Group0
    - Objects

Map B:
- Group1
    - Objects <-- this layer will be selected in order to copy objects
```

Swapping between `Map A -> Map B` and swapping back to `Map A` after selecting some objects from `Map B` it places me in `Group0/Objects` when I return to `Map A` instead of `Group1/Objects`, not that it matters because I had selected the layer `Group1/ActualTarget` before I left `Map A`, intending to paste there.

### Goal

I want to be able to edit multiple maps simultaneously with complex group structures and copy/paste between them onto different layers without having to constantly find the layer I was previously on.

1) Remember the open/closed status of group layers when switching maps
2) Remember the selected layer when swapping between maps
3) If none of the above, make a better guess at the layer including group structure?

**Comment by bjorn** [comment]
> 1. Remember the open/closed status of group layers when switching maps

This feature is currently missing.

> 2\. Remember the selected layer when swapping between maps

That's what Tiled does by default. The exception is when using a world to jump from one map to another, in which case it tries to be smart by keeping the same layer(s) selected, which it does by looking for layers with the same name.

> 3\. If none of the above, make a better guess at the layer including group structure?

I guess indeed the way forward would be to make the search for matching layers take into account group layers.

**Comment by UliAbo** [comment]
I don't how complex this is all to implement, but **point 1 "Remember the open/closed status of group layers when switching maps" would be very helpful** for me as I have a lot of groups and using at least 2 TMX files in parallel. And this drives me crazy opening the groups again after switching... 

Would it be possible to implement at least point 1 "soon-ish", as I could imagine the other two are more complex?

**Comment by bjorn** [comment]
@UliAbo Sure, I'll have a look at it.

**Comment by bjorn** [comment]
@UliAbo Done for Tiled 1.8.5 or 1.9, whichever comes first. You can [get a build from here](https://github.com/mapeditor/tiled/actions/runs/2089450912) in a moment. :-)

**Comment by UliAbo** [comment]
That's cool @bjorn 
I will give it a try today!

**Comment by UliAbo** [comment]
Isn't this implemented / done? Or am I wrong again? :D
I just tested it and it feels like all is in.

**Comment by bjorn** [comment]
> Isn't this implemented / done? Or am I wrong again? :D
> I just tested it and it feels like all is in.

Well, only "point 1" was implemented, but not "make the search for matching layers take into account group layers". So, when switching between maps in a world (so, not using the document tabs), currently the code synchronizing selected layers can still interfere in case you have multiple layers with the same name.

**Comment by kunal649** [comment]
hii @bjorn i'm taking a look at at the remaining points on the issue , specifically regarding how layers are selected when swapping between maps in a world. 
> I guess indeed the way forward would be to make the search for matching layers take into account group layers.

I initially tried implementing a fix for this (point # 3 making the search take into account group layers by building and comparing absolute paths like Group1/objects).
However, while testing, i realized that for the specific "copy/paste between maps" workflow mentioned in the original issue, any automatic layer synchronization (via switchToDocumentAndHandleSimiliarTileset) actively fights the user by overwriting their previously selected layer on the target map. To truly solve the copy/paste issue, it seems we might need an option to bypass the synchronization entirely when clicking between maps,like  falling back to a standard switchToDocument so each map retains its individual state. In video below i achieved exactly that results, by replacing the call with `switchToDocument` in `mouseReleaseEvent` within mapitem.cpp & worldmovemaptool.cpp. 
(whats your take on adding a toggle (e.g., "Sync layers when switching maps in World Tool") or is there something im missing!?)

https://github.com/user-attachments/assets/15701a32-1f71-406c-a44d-cbe3f9290bec

**Comment by bjorn** [comment]
@kunal649 You're right, the initial request here was to have an option of disabling the synchronization of selected layers, with the improvement of handling the group structure only suggested as a possible alternative. Such a toggle could certainly be useful!

**Comment by kunal649** [comment]
Nicee! I'm on it. I'll get that toggle into preferences and will be making a PR shortly.
Thanks (read my dm on discourse if time finds you :) )
